src/machine_helpers.py

```diff
@@ -170,12 +113,16 @@ def generate_service_args(auth: bool, machine_ip: str, config: MongoDBConfigurat
 def push_file_to_unit(parent_dir, file_name, file_contents) -> None:
     """K8s charms can push files to their containers easily, this is the vm charm workaround."""
     Path(parent_dir).mkdir(parents=True, exist_ok=True)
+    file_name = f"{parent_dir}/{file_name}"
     with open(file_name, "w") as write_file:
         write_file.write(file_contents)
 
-    os.chmod(file_name, 0o400)
+    if "keyFile" in file_name:
+        os.chmod(file_name, 0o400)
+    else:
+        os.chmod(file_name, 0o440)
```

Comment:
I think there some mistake
```suggestion
    if os.path.exists(file_name):
        os.chmod(file_name, 0o400)
```

Reply:
resolved per our conversation in mattermost. 

TL;DR is that it is necessary that the KeyFile has `400` for mongod to start, but for files like the TLS files it is necessary to have `440` so the user can verify TLS ops

Reply:
Can we add a code comment here? to explain the because of MongoDB limitation it is needed 400 rights for keyfile and we need 440 rights on tls certs to be able to connect via MongoDB shell